sdk/vaa/structs.go

```diff
@@ -645,21 +645,42 @@ func (v *BatchVAA) signingBody() []byte {
 	return buf.Bytes()
 }
 
-// SigningMsg returns the hash of the signing body.
-func SigningMsg(data []byte) common.Hash {
+func doubleKeccak(bz []byte) common.Hash {
 	// In order to save space in the solana signature verification instruction, we hash twice so we only need to pass in
 	// the first hash (32 bytes) vs the full body data.
-	return crypto.Keccak256Hash(crypto.Keccak256Hash(data).Bytes())
+	return crypto.Keccak256Hash(crypto.Keccak256Hash(bz).Bytes())
 }
 
-// SigningMsg returns the hash of the signing body. This is used for signature generation and verification
-func (v *VAA) SigningMsg() common.Hash {
-	return SigningMsg(v.signingBody())
+// This is a temporary method to produce a vaa signing digest on raw bytes.
+// It is error prone and we should use `v.SigningDigest()` instead.
+// whenever possible.
+// This will be removed in a subsequent release.
+func DeprecatedSigningDigest(bz []byte) common.Hash {
+	return doubleKeccak(bz)
 }
 
-// SigningMsg returns the hash of the signing body. This is used for signature generation and verification
-func (v *BatchVAA) SigningMsg() common.Hash {
-	return SigningMsg(v.signingBody())
+// MessageSigningDigest returns the hash of the data prepended with it's signing prefix.
+// This is intending to be used for signing messages of different types from VAA's.
+// The message prefix help protect from message collisions.
+func MessageSigningDigest(prefix []byte, data []byte) (common.Hash, error) {
+	if len(prefix) < 32 {
```

Comment:
I believe @tbjump said the check should be that the `len(prefix) + len(data) >= 34`.

Reply:
doesn't matter too much for now, we can come back to this if/once the p2p messages are ported over to use this function.